examples/gno.land/p/demo/grc/grc20/types.gno

```diff
@@ -55,9 +50,42 @@ type Token interface {
 	TransferFrom(from, to std.Address, amount uint64) error
 }
 
+var (
+	ErrInsufficientBalance   = errors.New("insufficient balance")
+	ErrInsufficientAllowance = errors.New("insufficient allowance")
+	ErrInvalidAddress        = errors.New("invalid address")
+	ErrCannotTransferToSelf  = errors.New("cannot send transfer to self")
+	ErrReadonly              = errors.New("banker is readonly")
+	ErrRestrictedTokenOwner  = errors.New("restricted to bank owner")
+)
+
 const (
 	MintEvent     = "Mint"
 	BurnEvent     = "Burn"
 	TransferEvent = "Transfer"
 	ApprovalEvent = "Approval"
 )
+
+type Token struct {
+	name     string
+	symbol   string
+	decimals uint
+	ledger   *AdminLedger
+}
+
+// AdminLedger is a struct that holds the balances and allowances and exposes
+// administrative functions.
+// It is unsafe to expose it publicly.
+type AdminLedger struct {
```

Comment:
Idea: what if we called this `UnsafeLedger`?

The point of the `Admin-` prefix is to make sure users don't export it, as far as I understand. maybe `Unsafe` or `Private` is better as a prefix.

Reply:
I like Private. It's my new Admin. :)

Reply:
a24c05ffd